drivers/memc/memc_stm32_xspi_psram_aps6404.c

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (c) 2025 Mario Paja
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdint.h>
+#define DT_DRV_COMPAT st_stm32_xspi_psram_aps6404
+
+#include <errno.h>
+#include <soc.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/multi_heap/shared_multi_heap.h>
+
+LOG_MODULE_REGISTER(memc_stm32_xspi_psram_aps6404, CONFIG_MEMC_LOG_LEVEL);
+
+#define STM32_XSPI_NODE DT_INST_PARENT(0)
+
+#ifdef CONFIG_SHARED_MULTI_HEAP
+struct shared_multi_heap_region smh_psram = {
+	.addr = DT_REG_ADDR(DT_NODELABEL(psram)),
+	.size = DT_REG_SIZE(DT_NODELABEL(psram)),
+	.attr = SMH_REG_ATTR_EXTERNAL,
+};
+#endif
+
+#define STM32_XSPI_CLOCK_PRESCALER_MIN                0U
+#define STM32_XSPI_CLOCK_PRESCALER_MAX                255U
+#define STM32_XSPI_CLOCK_COMPUTE(bus_freq, prescaler) ((bus_freq) / ((prescaler) + 1U))
+
+struct memc_stm32_xspi_psram_config {
+	const struct pinctrl_dev_config *pcfg;
+	const struct stm32_pclken pclken;
+	const struct stm32_pclken pclken_ker;
+	const struct stm32_pclken pclken_mgr;
+	size_t memory_size;
+	bool qspi_enable;
+	uint32_t max_frequency;
+};
+
+struct memc_stm32_xspi_psram_data {
+	XSPI_HandleTypeDef hxspi;
+};
+
+static int ap_memory_reset(XSPI_HandleTypeDef *hxspi)
+{
+	XSPI_RegularCmdTypeDef cmd = {0};
+
+	cmd.Instruction = 0x66;
+	cmd.InstructionMode = HAL_XSPI_INSTRUCTION_1_LINE;
+	cmd.InstructionWidth = HAL_XSPI_INSTRUCTION_8_BITS;
+
+	/* Configure the command */
+	if (HAL_XSPI_Command(hxspi, &cmd, HAL_XSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("XSPI write command failed");
+		return -EIO;
+	}
+	k_busy_wait(100);
+
+	cmd.Instruction = 0x99;
```

Comment:
Please consider moving these constants to macros with names, so that the commands are easy to understand when reading the driver code